lib/ui/painting/image_encoding.cc

```diff
@@ -57,12 +57,22 @@ void InvokeDataCallback(std::unique_ptr<DartPersistentValue> callback,
 
 sk_sp<SkImage> ConvertToRasterImageIfNecessary(sk_sp<SkImage> image,
                                                GrContext* context) {
-  if (context == nullptr) {
-    // The context was null (software rendering contexts) so the image is likely
-    // already a raster image. Nothing more to do.
+  SkPixmap pixmap;
+  if (image->peekPixels(&pixmap)) {
+    // This is already a raster image.
     return image;
   }
 
+  sk_sp<SkImage> raster_image = image->makeRasterImage();
```

Comment:
Nit: `if (auto raster_image = ...) { return raster_image; }` reads a bit better IMO and does not pollute the scope with a variable that has no use later.

Reply:
Add a comment here that says this will only work for non-x-context images and lazy images.

Reply:
done